keps/sig-network/2732-np-version/README.md

```diff
@@ -0,0 +1,803 @@
+  respect to NetworkPolicy features. Specifically:
+
+    - Allow the apiserver to indicate to the network plugin what
+      version of the NetworkPolicy featureset a particular policy
+      requires (eg, whether it uses combined `podSelector` and
+      `namespaceSelector`), so that network plugins can reliably
+      recognize when a NetworkPolicy makes use of a feature that the
+      NetworkPolicy implementation does not support.
+
+    - Allow NetworkPolicy implementations to indicate back to the user
+      when they are aware that they cannot correctly implement a given
+      NetworkPolicy.
+
+- Define rules for dealing with feature gates and alpha APIs in
+  NetworkPolicy that work well with the 3-way versioning split.
+
+- Allow network plugins to indicate when a NetworkPolicy has been fully
+  "programmed" into the cluster network, to allow clients (including
+  humans, scripts, and cluster components) to get more reliable behavior
+  when adding new policies.
+
+- (Maybe) Allow network plugins to provide some sort of global status
+  object with certain information about the implementation.
+
+### Non-Goals
+
+- Deprecating (or requiring) the use of feature gates to indicate the
+  stability / "certainty" of NetworkPolicy API additions; whether a
+  feature is API-stable or not is distinct from whether it is widely
+  implemented by plugins or not (though there is generally bidirectional
+  influence between the two).
+
+- Defining "metric-like" NetworkPolicy status information (eg, how long
+  a particular rule took to implement).
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - Testing Features of Different NetworkPolicy Implementations
+
+As a Kubernetes developer (who might be named Jay), I want to create a
+NetworkPolicy test suite, and test the NetworkPolicy implementations of
+various network plugins, but I don't want to have to manually keep track
+of which plugins (and which versions of which plugins) are expected to
+correctly implement which NetworkPolicy features.
+
+#### Story 2 - More Reliable NetworkPolicy Test Cases
+
+As a developer, I want NetworkPolicy e2e test cases that work reliably,
+meaning I want the test code to be able to tell at what point a
+newly-created NetworkPolicy is expected to be in effect, so that I don't
+time out the test case before the network plugin has managed to process
+the policy.
+
+#### Story 3 - Understanding Whether New Features Are Supported
+
+As a user, I want to be able to tell whether the cluster I am working
+in implements certain new NetworkPolicy features, such as
+[NetworkPolicy port ranges] or [selecting Namespaces by name].
+
+[NetworkPolicy port ranges]: https://github.com/kubernetes/enhancements/pull/2090
+[selecting Namespaces by name]: https://github.com/kubernetes/enhancements/pull/2113
+
+#### Story 4 - Reporting NetworkPolicy Version
+
+As a network plugin developer, I want to verify that a CNI implements a specific
+NetworkPolicy version.
```

Comment:
I don't understand what this means. Related: "A CNI" is not a thing. There is only one CNI, and it is [this](https://github.com/containernetworking/cni/blob/main/SPEC.md). (However, assuming you meant "a network plugin", the story still does not make sense.)